component/core/src/test/java/org/exoplatform/social/core/profile/ProfilePropertySettingsServiceTest.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * Copyright (C) 2023 Meeds Association contact@meeds.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.exoplatform.social.core.profile;
+
+import static org.junit.Assert.assertThrows;
+
+import org.exoplatform.commons.ObjectAlreadyExistsException;
+import org.exoplatform.social.core.jpa.storage.dao.jpa.ProfilePropertySettingDAO;
+import org.exoplatform.social.core.profile.settings.ProfilePropertySettingsService;
+import org.exoplatform.social.core.profileproperty.model.ProfilePropertySetting;
+import org.exoplatform.social.core.test.AbstractCoreTest;
+
+public class ProfilePropertySettingsServiceTest extends AbstractCoreTest {
+
+  private ProfilePropertySettingsService profilePropertySettingsService;
+
+  private ProfilePropertySettingDAO      profilePropertySettingDAO;
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    profilePropertySettingsService = getContainer().getComponentInstanceOfType(ProfilePropertySettingsService.class);
+    profilePropertySettingDAO = getContainer().getComponentInstanceOfType(ProfilePropertySettingDAO.class);
+  }
+
+  @Override
+  public void tearDown() throws Exception {
+    end();
+    begin();
+    profilePropertySettingDAO.deleteAll();
+    super.tearDown();
+  }
+
+  public void testCreateProfilePropertySetting() {
+    Throwable exception =
+            assertThrows(IllegalArgumentException.class,
+                    () -> this.profilePropertySettingsService.createPropertySetting(null));
+    assertEquals("Profile property setting Item Object is mandatory", exception.getMessage());
+    exception =
+            assertThrows(IllegalArgumentException.class,
+                    () -> this.profilePropertySettingsService.createPropertySetting(new ProfilePropertySetting()));
+    assertEquals("Profile property name is mandatory", exception.getMessage());
+    try {
+      ProfilePropertySetting profilePropertySetting = createProfileSettingInstance("test");
+      profilePropertySettingsService.createPropertySetting(profilePropertySetting);
+      profilePropertySetting = profilePropertySettingsService.getProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertNotNull(profilePropertySetting.getId());
+    } catch (Exception e) {
```

Comment:
please avoid using try/catch in tests, see above

Reply:
Done